appserver/web/web-core/src/main/java/org/apache/catalina/connector/AsyncContextImpl.java

```diff
@@ -644,6 +645,10 @@ void notifyAsyncListeners(AsyncEventType asyncEventType, Throwable t) {
                     AsyncEvent asyncEvent = new AsyncEvent(
                         this, asyncListenerContext.getRequest(),
                         asyncListenerContext.getResponse(), t);
+                    if(asyncListenerContext.getResponse() instanceof HttpServletResponse) {
+                        ((HttpServletResponse)asyncListenerContext.getResponse()).
+                                setStatus(HttpServletResponse.SC_OK);
+                    }
```

Comment:
I removed this block of code and I created a challenge because we identified that the wrong part is the TCK side, here the link of the challenge: https://github.com/jakartaee/cdi-tck/issues/389